lua/nvim-tree/git/init.lua

```diff
@@ -13,22 +16,37 @@ function M.reload()
   end
 
   for project_root in pairs(M.projects) do
-    M.projects[project_root] = {}
-    local git_status = Runner.run {
-      project_root = project_root,
-      list_untracked = git_utils.should_show_untracked(project_root),
-      list_ignored = true,
-      timeout = M.config.timeout,
-    }
-    M.projects[project_root] = {
-      files = git_status,
-      dirs = git_utils.file_status_to_dir_status(git_status, project_root),
-    }
+    M.reload_project(project_root)
   end
 
   return M.projects
 end
 
+function M.reload_project(project_root)
+  local project = M.projects[project_root]
+  if not project or not M.config.enable then
+    return
+  end
+
```

Comment:
We could move the debounce in here or maybe Runner::run 

That would debounce all the jobs, not just the .git ones.

That can come later.